src/metabase/mbql/util.clj

```diff
@@ -179,7 +179,7 @@
              [&match])
           subclauses))
 
-(s/defn simplify-compound-filter :- (s/maybe mbql.s/Filter)
+(defn simplify-compound-filter
```

Comment:
Explain

Reply:
Since we now have fliter-like clauses in :aggregation position as well, what gets returned from simplify-compound-filter will no longer necessarily be a filter. I could make it a sum type I suppose but the type would be kind of weird: either Filter or Query.

Reply:
... another option would be to make it either Filter or Aggregation and slightly rework the code that's using it

Reply:
Ok, so you're using this directly on aggregations now too? I never thought about using it that way. If it works that's great, by looking at it I can't think of anything off the top of my head that would break.

I would however humbly ask for some tests that it works in the weird new situations you're using it in

Reply:
Just a few for the new use cases

Reply:
It works out of the box. `mbql.util/replace` is super sweet piece of tech.

It's already implicitly tested in desugar-test, but those should be explicit, you're right.

Reply:
Oh yeah good point. I do maybe want to spin the MBQL stuff out into a separate lib one day maybe if we want to let people write drivers and plugins without using the entire Metabase project as a dependency so it would be good to have its own tests there.